pkg/network/master/subnets.go

```diff
@@ -55,15 +55,68 @@ func (master *OsdnMaster) SubnetStartMaster(clusterNetworks []common.ClusterNetw
 	}
 	master.subnetAllocatorList = subnetAllocatorList
 
+	err = master.verifyHostSubnetsState(subnets.Items)
+	if err != nil {
+		return err
+	}
+
 	master.watchNodes()
 	master.watchSubnets()
 	return nil
 }
 
+func (master *OsdnMaster) verifyHostSubnetsState(hostSubnets []networkapi.HostSubnet) error {
+	nodes, err := master.kClient.Core().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Error in initializing/fetching nodes: %v", err))
+		return err
+	}
+	nodeUIDmap := make(map[string]string)
+	for _, node := range nodes.Items {
+		nodeUIDmap[node.Name] = string(node.UID)
+	}
+	for _, subnet := range hostSubnets {
+		if nodeUIDmap[subnet.Name] == subnet.Annotations["nodeUID"] {
+			continue
+		}
+		if len(nodeUIDmap[subnet.Name]) > 0 {
+			//for backwards compatability - update hostsubnets created before we started tagging with nodes UID
+			if len(subnet.Annotations["nodeUID"]) == 0 {
+				newAnnotationsMap := map[string]string{
+					"nodeUID": nodeUIDmap[subnet.Name],
+				}
+				for key := range subnet.Annotations {
+					newAnnotationsMap[key] = subnet.Annotations[key]
+				}
+				subnet.Annotations = newAnnotationsMap
+				_, err := master.networkClient.Network().HostSubnets().Update(&subnet)
+				if err != nil {
+					return fmt.Errorf("error updating subnet %s for node %s: %v", subnet, subnet.Name, err)
+				}
+			} else if subnet.Annotations["nodeUID"] != nodeUIDmap[subnet.Name] {
```

Comment:
Can we have a comment explaining this whole thing?  It would be helpful for the future.  Make it clear when the two can get out of sync.

Reply:
To be honest I am not sure when they could get out of sync, besides someone editing the annotation. When I force this branch to be taken the system automatically creates a new hostsubnet for the node

Reply:
They could be out of sync in this case:
1. Node got added
2. (1) triggers creation of HostSubnet with uid annotation
3. Node got deleted but the event got missed, so HostSubnet still exists 
4. New node got added with the same name as (1)
5. If you are processing the subnet created in (2), you will find an entry in nodeUIDmap but the uid annotations will not match.